content/blog/2022/01/multiplatform-kubewarden.md

```diff
@@ -0,0 +1,76 @@
+---
+title: Multiplatform Kubewarden
+authors:
+    - Rafael Fernández López
+date: 2022-02-01
+---
+
+The Kubewarden team is glad to announce that in the spirit of helping
+Policy Authors and Cluster Administrators, the project is now
+officially multiplatform.
+
+The list of supported platforms as of now are:
+
+- Policy Server, as [container image](https://github.com/kubewarden/policy-server/pkgs/container/policy-server):
+  - `linux/amd64` (with the `musl` libc)
+  - `linux/arm64` (with the `musl` libc)
+
+- kwctl, as a [standalone binary](https://github.com/kubewarden/kwctl/releases):
+  - `darwin` (`x86_64`)
+  - `linux` (`aarch64`, with the `musl` libc)
+  - `linux` (`x86_64`, with the `musl` libc)
+  - `windows` (`x86_64`, with `MSVC`)
+
+We have tried to prioritize the same toolchain on platforms where we
+were able to do so.
+
+## Looking for platform convergence
+
+There are two main dependencies in the previous project layout that
+are somewhat special and sometimes problematic to converge to. One is
+`glibc`, and the other is `openssl`.
+
+### The libc
+
+`glibc` only applies to `linux` builds, so in principle is not what
+keeps us from multi platform convergence. However, we thought that
+producing a final static binary for our users in all platforms is
+interesting; specially for a CLI tool like `kwctl`, that our users
+will interact with directly in their environment.
```

Comment:
```suggestion
`glibc` is only used for the `linux` builds, so it would not keep us from building multi platform binaries. And we think that producing a "static binary" for the different platforms is interesting; specially for a CLI tool like `kwctl`, which our users can install directly in their environments without caring about dependencies.
```

Reply:
What I try to mean here is that using glibc vs musl is not helping with other platforms other than Linux. It just helps with having a standalone binary one can execute directly.

What it helps with is to have this "same" UX in all platforms, Linux included, of just downloading a binary and execute it right away.

Reply:
how about listing more specific gains from the changes, besides reducing dependency graph complexity?

I would love a one-liner on how using musl allows us to cross-compile to several architectures, reducing compile time from hours in a virtualized environment to minutes when cross-compiling.

Reply:
Closing as discussed.